### PR TITLE
Kelly branch

### PR DESCRIPTION
**Pull request by Osoba001**

### CODE REVIEW

File: CCSANoteApp.DB/Repositories/IRepository.cs [hunk 1]
```diff
@@ -9,12 +9,12 @@ namespace CCSANoteApp.DB.Repositories
...
     public interface IRepository<T> where T : BaseEntity
     {
-        void Add(T obj);
-        T? GetById(Guid id);
-        List<T> GetAll();
-        void Update(T obj);
-        void Delete(T obj);
-        void DeleteById(Guid id);
-        
+        Task<bool> Add(T entity);
```
Comment (IruomaJennifer) [inline CCSANoteApp.DB/Repositories/IRepository.cs:12]: Why are you using Task<bool> and what is the implication of not using it?

File: CCSANoteApp.DB/Repositories/Repository.cs [hunk 2]
```diff
@@ -10,61 +11,61 @@ namespace CCSANoteApp.DB.Repositories
...
-        public List<T> GetAll()
+        public async Task<bool> DeleteById(Guid id)
         {
-            var collection = _session.Query<T>().ToList();
-            return collection;
+            var ent = await _session.Query<T>().FirstOrDefaultAsync(x => x.Id == id);
+            if (ent != null)
+            {
+                await _session.DeleteAsync(ent);
+                return await Commit();
```
Comment (IruomaJennifer) [inline CCSANoteApp.DB/Repositories/Repository.cs:39]: I didn't know that Commit() is awaitable

File: CCSA_Web/Program.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@
 builder.Services.AddEndpointsApiExplorer();
 builder.Services.AddSwaggerGen();
 
-//builder.Services.AddSingleton(builder.Configuration.GetSection(nameof(DBConfiguration)).Get<DBConfiguration>());
+builder.Services.AddSingleton<SessionFactory>(x => new SessionFactory(builder.Configuration.GetConnectionString("defaultConnection")));
```
Comment (IruomaJennifer) [inline CCSA_Web/Program.cs:15]: I don't understand this line

**Review by IruomaJennifer — APPROVED** [review]

File: CCSA_Web/Program.cs (hunk 3 above)
Comment (Osoba001) [reply CCSA_Web/Program.cs:15]: I am getting the connection string from my AppSettings and passing it to SessionFactory class through constructor. i will explain in detail later if it is not clear enough

**Pull request merged**